kale/embed/mogonet.py

```diff
@@ -0,0 +1,130 @@
+# =============================================================================
+# Author: Sina Tabakhi, sina.tabakhi@gmail.com
+# =============================================================================
+
+"""
+Construct a message passing network using PyTorch Geometric for the MOGONET method. MOGONET is a multiomics fusion
+framework for cancer classification and biomarker identification that utilizes supervised graph convolutional networks
+for omics datasets.
+
+This code is written by refactoring MOGONET code (https://github.com/txWang/MOGONET/blob/main/models.py) within
+'MessagePassing' base class provided in the PyTorch Geometric.
+
+Reference:
+Wang, T., Shao, W., Huang, Z., Tang, H., Zhang, J., Ding, Z., Huang, K. (2021). MOGONET integrates multi-omics data
+using graph convolutional networks allowing patient classification and biomarker identification. Nature communications.
+https://www.nature.com/articles/s41467-021-23774-w
+"""
+
+from typing import List, Optional, Union
+
+import torch
+import torch.nn.functional as F
+import torch_sparse
+from torch import Tensor
+from torch.nn import Module, Parameter
+from torch.nn.init import xavier_normal_
+from torch_geometric.nn.aggr import Aggregation
+from torch_geometric.nn.conv import MessagePassing
+from torch_sparse import SparseTensor
+
+
+class MogonetGCNConv(MessagePassing):
+    r"""Create message passing layers for the MOGONET method. Each layer is defined as:
+
+    .. math::
+        H^{(l+1)}=f(H^{(l)}, A) = \sigma(AH^{(l)}W^{(l)})
+
+    where :math:`\mathbf{H^{(l)}}` is the input of the :math:`l` th layer and :math:`\mathbf{W^{(l)}}` is the weight
+    matrix of the :math:`l` th layer. :math:`\sigma(.)` denotes a non-linear activation function.
+
+    For more information please refer to the
```

Comment:
Do we have to repeat this when you've given global reference at the top?

Reply:
The duplicate reference has been removed.